client/blocks/site-launch-gate/index.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize, LocalizeProps } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { localizeUrl } from 'lib/i18n-utils';
+import { getSectionName } from 'state/ui/selectors';
+import getSelectedSiteId from 'state/ui/selectors/get-selected-site-id';
+import { launchSite } from 'state/sites/launch/actions';
+import Card from 'components/card';
+import ButtonGroup from 'components/button-group';
+import Button from 'components/button';
+import InlineSupportLink from 'components/inline-support-link';
+import isUnlaunchedSite from 'state/selectors/is-unlaunched-site';
+
+interface Props {
+	isUnlaunched: boolean;
+	launchSite: Function;
+	section: string;
+	siteId?: number;
+}
+
+class SiteLaunchGate extends Component< Props & LocalizeProps > {
+	render() {
+		const { children, isUnlaunched, section, siteId, translate } = this.props;
+
+		if ( isUnlaunched ) {
+			return <>{ children }</>;
+		}
+
+		return (
+			<Card>
+				{ section === 'themes'
+					? translate( 'To continue installing the theme, you must launch your site.' )
+					: translate( 'To continue installing the plugin, you must launch your site.' ) }
+				&nbsp;
+				<InlineSupportLink
+					supportPostId={ 1507 }
+					supportLink={ localizeUrl(
+						'https://en.support.wordpress.com/settings/privacy-settings/'
+					) }
+				/>
+				<br />
+				<ButtonGroup>
+					<Button
+						onClick={ e => {
+							e.preventDefault();
+							// TODO
+						} }
+					>
+						{ translate( 'Launch (but hide from Search Engines)' ) }
+					</Button>
+					<Button
+						primary
+						onClick={ e => {
+							e.preventDefault();
+							this.props.launchSite( siteId );
+						} }
+					>
+						{ translate( 'Launch Site' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Launch site' )` *ES Score: 11*